File: 2023/Global and Local Inversions.java
class Solution {
    public boolean isIdealPermutation(int[] nums) {
        int max=-1;
        for(int i=0; i<nums.length-2; i++){
            max=Math.max(max, nums[i]);
            if(max>nums[i+2])return false;
        }
        return true;
    }
}

File: 2023/Global and Local Inversions.cpp
class Solution {
public:
    bool isIdealPermutation(vector<int>& nums) {
        int n=nums.size(), local=0, global=0;
        for(int i = 0; i<n-1; i++){
            if(nums[i] > nums[i + 1])local++;
        }
        for(int i=0; i<n; i++){
            for(int j=i+1; j<n; j++){
                if(nums[i]>nums[j])global++;
            }
        }
        if(local==global)return true;
        return false;
    }
};

//o(n)

class Solution {
public:
    bool isIdealPermutation(vector<int>& nums) {
        int n=nums.size(), maxi=-1;
        for(int i=0; i<n-2;i++){
            maxi=max(maxi, nums[i]);
            if(maxi>nums[i+2])return false;
        }
        return true;
    }
};
